planty/chats/views.py

```diff
@@ -0,0 +1,102 @@
+from django.shortcuts import redirect, render
+from .models import Message, Chat
+from .forms import MessageForm
+from django.utils import timezone
+from posts.models import Post
+from django.db.models import Q
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpResponseNotFound
+from .utils import return_list_last_messages_in_chat
+
+
+@login_required
+def first_message(request, pk):
+    try:
+        post_instance = Post.objects.get(id=pk)
+    except ObjectDoesNotExist:
+        return HttpResponseNotFound()
+
+    try:  # Chat already exists
+        chat = Chat.objects.get(Q(buyer=request.user) & Q(post=post_instance))
+        return redirect('chat', pk=chat.id)
+    except ObjectDoesNotExist:
+        form = MessageForm()
+        if request.method == "POST":
+            form = MessageForm(request.POST)
+            if form.is_valid():
+                chat = Chat(post=post_instance, buyer=request.user)
+                chat.save()
+                message = form.save(commit=False)
+                message.date = timezone.now()
+                message.chat = chat
+                message.read_message_seller = False
+                message.sender = request.user
+                message.save()
+                return redirect('chat', pk=chat.id)
+            else:
+                messages.error(
+                    request, "Your message wasn't sent. Please try again.")
+
+        context = {
+            'form': form,
+            'post': post_instance,
+        }
+        return render(request, 'chats/first_message.html', context)
+
+
+@login_required
+def chat(request, pk):
+    try:
+        relevant_chat = Chat.objects.get(Q(id=pk) & (
+            Q(buyer=request.user) | Q(post__user=request.user)))
+    except ObjectDoesNotExist:
+        return HttpResponseNotFound()
+
+    buyer = relevant_chat.buyer
+    post = relevant_chat.post
```

Comment:
Nothing wrong with this but I would personally just remove these two lines and use `chat.buyer` etc. Up to you if you want to change it or not :)